hibernate-core/src/main/java/org/hibernate/type/descriptor/sql/internal/JdbcLiteralFormatterCharacterData.java

```diff
@@ -34,12 +37,15 @@ public JdbcLiteralFormatterCharacterData(JavaTypeDescriptor javaTypeDescriptor,
 	public String toJdbcLiteral(Object value, Dialect dialect, SharedSessionContractImplementor session) {
 		final String literalValue = unwrap( value, String.class, session );
 
+		final String inlineLiteral = dialect.inlineLiteral( literalValue );
+
 		if ( isNationalized ) {
 			// is there a standardized form for n-string literals?  This is the SQL Server syntax for sure
-			return String.format( Locale.ROOT, "n'%s'", literalValue );
+			return String.format( Locale.ROOT, NATIONALIZED_FORMAT, inlineLiteral );
+
 		}
 		else {
-			return String.format( Locale.ROOT, "'%s'", literalValue );
+			return String.format( Locale.ROOT, FORMAT, inlineLiteral );
```

Comment:
Any difference between
```
String.format( Locale.ROOT, "%s", inlineLiteral );
```
and
```
inlineLiteral
```

A little bit curious for I think this might be one of my knowledge gaps
```

Reply:
good point, I think we can avoid `return String.format( Locale.ROOT, FORMAT, inlineLiteral );` and just return `inlineLiteral`